Python/ceval.c

```diff
@@ -1759,18 +1789,45 @@ _PyEvalFramePushAndInit(PyThreadState *tstate, PyFunctionObject *func,
     Py_DECREF(func);
     Py_XDECREF(locals);
     for (size_t i = 0; i < argcount; i++) {
-        Py_DECREF(args[i]);
+        PyStackRef_CLOSE(args[i]);
     }
     if (kwnames) {
         Py_ssize_t kwcount = PyTuple_GET_SIZE(kwnames);
         for (Py_ssize_t i = 0; i < kwcount; i++) {
-            Py_DECREF(args[i+argcount]);
+            PyStackRef_CLOSE(args[i+argcount]);
         }
     }
     PyErr_NoMemory();
     return NULL;
 }
 
+static _PyInterpreterFrame *
+_PyEvalFramePushAndInit_UnTagged(PyThreadState *tstate, PyFunctionObject *func,
```

Comment:
Would it be more efficient, and maybe simpler to do the work of converting to stack refs in `_PyEval_Vector` and `_PyEvalFramePushAndInit_Ex`?

Reply:
I tried, and it's actually a bit messier because we need to convert twice. And _PyEvalFramePushAndInit needs stackrefs anyways.